modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.loader.csv;
+
+import static com.opengamma.strata.product.common.PayReceive.PAY;
+
+import java.time.LocalDate;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.CurrencyAmount;
+import com.opengamma.strata.collect.io.CsvRow;
+import com.opengamma.strata.loader.LoaderUtils;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.common.PayReceive;
+import com.opengamma.strata.product.fx.FxSingle;
+import com.opengamma.strata.product.fx.FxSingleTrade;
+
+/**
+ * Loads FX trades (spot of forward) from CSV files.
+ */
+class FxSingleTradeCsvLoader {
+  private static final String PAYMENT_DATE_HEADER = "Payment Date";
+  private static final String LEG_1_DIRECTION_HEADER = "Leg 1 Direction";
+  private static final String LEG_1_CURRENCY_HEADER = "Leg 1 Currency";
+  private static final String LEG_1_NOTIONAL_HEADER = "Leg 1 Notional";
+  private static final String LEG_2_DIRECTION_HEADER = "Leg 2 Direction";
+  private static final String LEG_2_CURRENCY_HEADER = "Leg 2 Currency";
+  private static final String LEG_2_NOTIONAL_HEADER = "Leg 2 Notional";
+
+  /**
+   * Parses the data from a CSV row.
+   *
+   * @param row  the CSV row object
+   * @param info  the trade info object
+   * @param resolver  the resolver used to parse additional information. This is not currently used in this method.
+   * @return the parsed trade, as an instance of {@link FxSingleTrade}
+   */
+  static FxSingleTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    FxSingleTrade trade = parseRow(row, info, resolver);
+    return resolver.completeTrade(row, trade);
+  }
+
+  /**
+   * Parses the data from a CSV row.
+   *
+   * @param row  the CSV row object
+   * @param info  the trade info object
+   * @param resolver  the resolver used to parse additional information. This is not currently used in this method.
+   * @return the parsed trade, as an instance of {@link FxSingleTrade}
+   */
+  private static FxSingleTrade parseRow(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    LocalDate paymentDate = LoaderUtils.parseDate(row.getField(PAYMENT_DATE_HEADER));
```

Comment:
Should also support business day adjustment fields as per a swap.

Reply:
@marc-henrard How do you suggest we add this to the CSV format (header)?

Reply:
See `FullSwapCsvLoader`.

```
  private static final String PAYMENT_DATE = "Payment Date";
  private static final String PAYMENT_DATE_CNV_FIELD = "Payment Date Convention";
  private static final String PAYMENT_DATE_CAL_FIELD = "Payment Date Calendar";
```